session/src/main/scala/akka/cassandra/session/scaladsl/CassandraSession.scala

```diff
@@ -64,6 +65,15 @@ final class CassandraSession(
     init(session).map(_ => session)
   }
 
+  def this(
+      system: ActorSystem,
+      sessionProvider: CqlSessionProvider,
+      executionContext: ExecutionContext,
+      log: LoggingAdapter,
+      metricsCategory: String,
+      init: CqlSession => Future[Done]) =
```

Comment:
maybe we don't need both constructor variants. creating a session with `new` is anyway very power user api